frontend/src/services/Api.ts

```diff
@@ -106,21 +106,10 @@ const getHeaders = (): Partial<AxiosHeaders> => {
   if (apiProxy) {
     return { 'Content-Type': 'application/x-www-form-urlencoded' };
   } else {
-    return { ...loginHeaders };
+    return { 'Content-Type': 'application/json', ...loginHeaders };
   }
 };
 
-/** Create content type correctly for a given request type */
-const getHeadersWithMethod = (method: HTTP_VERBS): Partial<AxiosHeaders> => {
-  let allHeaders = getHeaders();
-
-  if (method === HTTP_VERBS.PATCH) {
```

Comment:
This is not valid because there are POST methods that send JSON body and they should have header `Content-Type: application/json`. One example is `getMetricsStats`. 

```
export const getMetricsStats = (queries: MetricsStatsQuery[]): Promise<ApiResponse<MetricsStatsResult>> => {
  return newRequest<MetricsStatsResult>(HTTP_VERBS.POST, urls.metricsStats, {}, { queries: queries });
};
```

Since all API requests send JSON body (`{}` for empty requests), we can add this header to all requests and avoid HTTP method distinction.